### DISCUSSION TITLE
Need an example of user login auth page ( Basic authentication login)

### DISCUSSION THREAD

**Discussion by cary-sas** [observablehq/framework#1554] [Q&A] [answered]
My project need a user login auth page, since different user will assgin them diffferent permissions.
Please give us more examples of user login auth.

**Comment by thierrymoudiki** [comment]
@thierrymoudiki

**Reply by jaanaltosaar** [reply]
We need this too @onefact

**Comment by oliveryh** [comment]
Until/if this feature is added to Observable Framework, in case anyone here is trying to do this and is using NGINX, I was able to satisfy my immediate need for basic auth using [this guide](https://www.digitalocean.com/community/tutorials/how-to-set-up-password-authentication-with-nginx-on-ubuntu-14-04).

**Comment by mbostock** [comment, marked as answer]
Since Framework is a static site generator, you implement authentication and access control when serving your static site.

We recommend using [Observable Cloud](https://observablehq.com/platform/cloud) for hosting private Framework apps. To control access to your site, you can either [invite users to your app](https://observablehq.com/documentation/data-apps/sharing), [create a secret link](https://observablehq.com/documentation/data-apps/sharing#secret-links-beta), or use on-premises serving and integrate with your own identity provider via OIDC.

In addition, Observable Cloud supports [segmentation](https://observablehq.com/documentation/data-apps/segmentation) which allows you to build a different version of your app — typically different data — for each user, thus serving personalized copies of your app for each user.
